### ISSUE TITLE
error: namespace "thrust::cub" has no member "DebugSyncStream" 

### ISSUE THREAD

**Issue by pzc163** [facebookresearch/pytorch3d#1610]
Windows10 
CUDA11.8 
pytorch=2.0.1

when i build pytorch3D in win10，i meet below this error:

D:/Program Files/Anaconda3/envs/torch3D/lib/site-packages/torch/include\c10/macros/Macros.h(138): warning C4067: 预处理器指令后有意外标记 - 应输入换行符
C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v11.8/include\thrust/system/cuda/detail/core/agent_launcher.h(524): error: namespace "thrust::cub" has no member "DebugSyncStream"

1 error detected in the compilation of "D:/Program Files/Anaconda3/envs/torch3D/Lib/site-packages/pytorch3d-0.7.4/pytorch3d/csrc/iou_box3d/iou_box3d.cu".
error: command 'C:\\Program Files\\NVIDIA GPU Computing Toolkit\\CUDA\\v11.8\\bin\\nvcc.exe' failed with exit code 2

(torch3D) D:\Program Files\Anaconda3\envs\torch3D\Lib\site-packages\pytorch3d-0.7.4>
![image](https://github.com/facebookresearch/pytorch3d/assets/105222039/08e435c1-d3cd-496b-b36c-69f1ea89f889)

**Comment by pzc163** [comment]
I ask New bing for this issue，new bing's response is  The error message “namespace “thrust::cub” has no member “DebugSyncStream”” usually occurs when you are trying to use DebugSyncStream in a version of Thrust that does not support it. [This is because DebugSyncStream was added in Thrust 1.9.0 and is not available in earlier versions](https://github.com/facebookresearch/pytorch3d/issues/1604)

If you are using an earlier version of Thrust, you can try upgrading to version 1.9.0 or later to use DebugSyncStream

can anyone help to confirm  that new bing's response is correct？

**Comment by bottler** [comment]
I don't know the specifics. There have been a few issues like this very recently. The only one with a fix is https://github.com/facebookresearch/pytorch3d/issues/1603 where they downgraded to cuda 11.7. Can you do that?

**Comment by pzc163** [comment]
> I don't know the specifics. There have been a few issues like this very recently. The only one with a fix is #1603 where they downgraded to cuda 11.7. Can you do that?

thanks for your reply bottler，I noticed that some people have solved this problem by downgrading CUDA to 11.7. But considering other projects, I really don’t want to downgrade the CUDA version to 11.7, so I am looking for other solutions

**Comment by lucasjinreal** [comment]
It's really hard to suggest users to downgrade cuda version to support pytorch3d, this is not the right way nor the proper way.

CUDA is much more basic in whole system for user, tons of projects, software, game dependes on it, 
Why not just listen voice from community and take a person's effort just look at this issue and fix CUDA11.8 build on windows?

2 Years, I didn't saw this issue resolve......

**Comment by bottler** [comment]
@lucasjinreal Not sure what you mean by "2 years". Internally we aren't using PyTorch3D on windows, and we haven't ever distributed binaries for Windows or had a working CI setup for windows. We think there are people using PyTorch3D successfully on windows, and we think that's required considerable effort from some of them them. We have tried to support users who report problems with Windows builds and we've made requested changes to help them, and we do hear of successes where these things have helped. This particular issue, referencing "DebugSyncStream", is very new - mentioned in 4 different issues since August 1 2023 and never before that, and we don't have the means to solve it, and of course we would appreciate more information on how it might be solved. It looks like #1509 succeeded in the end to build with CUDA 11.8 on windows, so I don't think CUDA 11.8 on windows is completely broken. Rather I think something more specific, possibly recent, is the problem here.

**Comment by GMPHL** [comment]
i  have the same problem

**Comment by lucasjinreal** [comment]
Yeah, but I suggest you downgrade cuda, this is how bottler suggested me....

**Comment by GMPHL** [comment]
> 

i downgrade the cuda to 11.7 on window. But it got error again. pytorch3d seem not support win....

**Comment by lucasjinreal** [comment]
Yeah, **this the only lib that suggest users downgrade cuda**, even though cuda might evolved hundredes of APPs based on it, still pytorch3d suggest users downgrade cuda. But not try fix their build errors....

IMO, just get rid of pytorch3d.

**Comment by lucasjinreal** [comment]
I have to correct this statement:

> ~pytorch3d don't support Windows.

It should be:

> pytorch3d build fail on Windows only under CUDA11.8 (or might above)

Please lead users to the correct direction....

**Comment by GMPHL** [comment]
> > i downgrade the cuda to 11.7 on window. But it got error again. pytorch3d seem not support win....
> 
> Indeed, we don't support Windows. Since the beginning, users on Windows have been helping each other.

Please, many people need pytorch3d on windows. Thx!~

**Comment by bottler** [comment]
I wonder if the recent commit to main, https://github.com/facebookresearch/pytorch3d/commit/6f2212da46f3ad1a596b3e1017be2d16eaaf95f9, might help with this?

**Comment by Sqiuln** [comment]
it works,thanks
